src/cdk-experimental/listbox/listbox.ts

```diff
@@ -7,98 +7,206 @@
  */
 
 import {
+  AfterContentInit,
   ContentChildren,
   Directive,
-  EventEmitter, forwardRef, Inject,
-  Input, Output,
+  ElementRef, EventEmitter, forwardRef,
+  Inject,
+  Input, OnDestroy, Output,
   QueryList
 } from '@angular/core';
+import {ActiveDescendantKeyManager, Highlightable, ListKeyManagerOption} from '@angular/cdk/a11y';
+import {ENTER, SPACE} from '@angular/cdk/keycodes';
 import {BooleanInput, coerceBooleanProperty} from '@angular/cdk/coercion';
 
 let nextId = 0;
 
-/**
- * Directive that applies interaction patterns to an element following the aria role of option.
- * Typically meant to be placed inside a listbox. Logic handling selection, disabled state, and
- * value is built in.
- */
 @Directive({
   selector: '[cdkOption]',
   exportAs: 'cdkOption',
   host: {
     role: 'option',
     '(click)': 'toggle()',
-    '[attr.aria-selected]': 'selected || null',
+    '(focus)': 'activateOption()',
+    '(blur)': 'deactivateOption()',
+    '[attr.aria-selected]': '_selected || null',
     '[id]': 'id',
+    '[attr.tabindex]': '_getTabIndex()',
+    '[attr.aria-disabled]': '_getComboDisabled()',
+    '[class.cdk-option-disabled]': '_getComboDisabled()',
+    '[class.cdk-option-active]': '_active'
+
   }
 })
-export class CdkOption {
+export class CdkOption implements ListKeyManagerOption, Highlightable {
   private _selected: boolean = false;
+  private _disabled: boolean = false;
+  _active: boolean = false;
 
-  /** Whether the option is selected or not */
   @Input()
   get selected(): boolean {
     return this._selected;
   }
   set selected(value: boolean) {
-    this._selected = coerceBooleanProperty(value);
+    if (!this._disabled) {
+      this._selected = coerceBooleanProperty(value);
+    }
   }
 
   /** The id of the option, set to a uniqueid if the user does not provide one */
   @Input() id = `cdk-option-${nextId++}`;
 
-  constructor(@Inject(forwardRef(() => CdkListbox)) public listbox: CdkListbox) {}
+  @Input()
+  get disabled(): boolean {
+    return this._disabled;
+  }
+  set disabled(value: boolean) {
+    this._disabled = coerceBooleanProperty(value);
+  }
+
+  constructor(private _el: ElementRef,
+              @Inject(forwardRef(() => CdkListbox)) public listbox: CdkListbox) {
+  }
 
   /** Toggles the selected state, emits a change event through the injected listbox */
   toggle() {
-    this.selected = !this.selected;
-    this.listbox._emitChangeEvent(this);
+    if (!this._getComboDisabled()) {
+      this.selected = !this.selected;
+      this.listbox._emitChangeEvent(this);
+    }
+  }
+
+  activateOption() {
+    if (!this._getComboDisabled()) {
+      this._active = true;
+      this.listbox.setActiveOption(this);
+    }
+  }
+
+  deactivateOption() {
+    this._active = false;
+  }
+
+  _getComboDisabled(): boolean {
+    return (this.listbox.disabled || this._disabled);
+  }
+
+  _getTabIndex(): string | null {
+    return (this.listbox.disabled || this._disabled) ? null : '-1';
```

Comment:
I made it a method because it was consistent with the _isInteractionDisabled() references in the other host bindings. Made the host section cleaner. If you think it would be better to remove the standalone method though I can do that.

Reply:
It's fine either way; my own preference would be to do these as expressions in the host block to keep the class smaller, but it's not a big deal either way